mesonbuild/modules/python2.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2018 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .. import dependencies
+from .python_common import PythonModule
+
+
+class Python2Module(PythonModule):
+    def _find_python(self):
+        return dependencies.ExternalProgram('python2', silent = True)
```

Comment:
This does not work on platforms where Python 2 has the name `python`. This includes at least macOS.

Reply:
Right, but that is IMO a larger problem with how the python 3 module currently works, we need the equivalent of the autotools macro for finding python, along with some sort of option that lets the user specify python's name comparable to how $PYYTHON was handled, at least that's how I see it

Reply:
I will certainly look at that in the coming days by the way, I just figured this commit worked standalone and could be proposed right away

Reply:
imo it would be nice if the API was something like

```python
pymod = import('python')

py = pymod.create('/some/path/or/just/binary/name/defaulting/to/python3')

py.extension_module(...)
py.language_version(...)
py.find_python(...)
...
```

Reply:
Yes something like that would be sweet, however it kind of conflicts with the existing, not sure how to go about that, @jpakkane I assume this would involve deprecating usage of the python3 module ?

Reply:
or find_python could take a param and as a side effect set the interpreter for all other future python module calls.

Reply:
That seems prone to breakage, I really like your first suggestion, will give it a go :)

Reply:
> as a side effect set the interpreter for all other future python module calls

No. No hidden global state. And this would prevent you from building Python 2 and 3 modules at the same time (which IIRC some people still need to support).